test/test_utils.py

```diff
@@ -225,3 +225,24 @@ def test_invalid_png_data(self):
                           imageinfo.identify, data)
         self.assertRaises(imageinfo.UnrecognizedFormat,
                           imageinfo.identify, data)
+
+class TestCrypt(unittest.TestCase):
+
+    def test_encode_filename(self):
+        self.assertEqual(b'\xe5\xb0\x8f\xe5\xae\xa4', util.encode_filename(u"小室"))
+        self.assertEqual(b'\xe5\xb0\x8f', util.encode_filename(b'\xe5\xb0\x8f'))
+
+        currentSupport = os.path.supports_unicode_filenames = True
+        self.assertEqual(u"小室", util.encode_filename(u"小室"))
+
+        currentPlatform = sys.platform
+        sys.platform = 'darwin'
+        self.assertEqual(b'\xe5\xb0\x8f\xe5\xae\xa4', util.encode_filename(u"小室"))
+        sys.platform = currentPlatform
+        os.path.supports_unicode_filenames = currentSupport
+
+
+    def test_decode_filename(self):
+        self.assertEqual('\u9061abc', util.decode_filename(b'\xe9\x81\xa1abc'))
+        self.assertEqual('Пётр Ильич Чайковский', util.decode_filename('Пётр Ильич Чайковский'))
+        self.assertEqual(u"小室哲哉", util.decode_filename(u"小室哲哉".encode()))
```

Comment:
This is python 3. You want to remove the u prefix. Also read more about cross platform encoding. Unicode won't always be represented by the same byte code on Linux vs windows.